x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherService.java

```diff
@@ -232,6 +233,7 @@ private synchronized void reloadInner(ClusterState state, String reason, boolean
      */
     public void pauseExecution(String reason) {
         int cancelledTaskCount = executionService.pause();
+        triggerService.pauseExecution();
```

Comment:
Does it maybe make sense to pause the trigger service before pausing the execution service?

Reply:
indeed. changed